app/api/resources/metadata.py

```diff
@@ -238,3 +239,23 @@ async def get_genome_dataset_attributes(
     except Exception as ex:
         logging.error(ex)
         return response_error_handler({"status": 500})
+
+@router.get("/genomeid")
+async def get_genome_by_keyword(request: Request, assembly_accession_id: str):
+    try:
+        genome_response = grpc_client.get_genome_by_specific_keyword(assembly_accession_id=assembly_accession_id)
+        latest_genome_by_keyword_object = GenomeByKeyword()
+        for arr in genome_response:
+            arr = MessageToDict(arr)
+            genome_by_keyword_object = GenomeByKeyword(**arr)
+            if (genome_by_keyword_object.release_version > latest_genome_by_keyword_object.release_version):
+                latest_genome_by_keyword_object = genome_by_keyword_object
+        if (latest_genome_by_keyword_object.genome_uuid):
+            return latest_genome_by_keyword_object
```

Comment:
Shouldn't it return a response (e.g. `responses.JSONResponse`) instead of a pydantic model isntance?

Reply:
I was doing it at the resolver end. Makes sense to do it here. Will update.